src/sentry/metrics/minimetrics.py

```diff
@@ -9,13 +9,13 @@
 
 def _to_minimetrics_external_metric_tags(tags: Optional[Tags]) -> Optional[MetricTagsExternal]:
     # We remove all `None` values, since then the types will be compatible.
-    casted_tags = None
+    casted_tags: Any = None
     if tags is not None:
         casted_tags = {
             tag_key: str(tag_value) for tag_key, tag_value in tags.items() if tag_value is not None
         }
 
-    return cast(Optional[MetricTagsExternal], casted_tags)
+    return casted_tags
```

Comment:
Apparently `Optional[MetricsTagsExternal]` causes something odd to happen in `typing` and this also raise a recursion error for me:

```
11:59:28 [ERROR] sentry.errors: Unable to record backend metric
Traceback (most recent call last):
  File "/Users/mitsuhiko/Development/sentry/src/sentry/utils/metrics.py", line 133, in incr
    backend.incr(key, instance, tags, amount, sample_rate)
  File "/Users/mitsuhiko/Development/sentry/src/sentry/metrics/middleware.py", line 133, in incr
    return self.inner.incr(key, instance, current_tags, amount, sample_rate)
  File "/Users/mitsuhiko/Development/sentry/src/sentry/metrics/minimetrics.py", line 86, in incr
    tags=_to_minimetrics_external_metric_tags(tags),
  File "/Users/mitsuhiko/Development/sentry/src/sentry/metrics/minimetrics.py", line 18, in _to_minimetrics_external_metric_tags
    return cast(Optional[MetricTagsExternal], casted_tags)
  File "/Users/mitsuhiko/.pyenv/versions/3.8.12/lib/python3.8/typing.py", line 258, in inner
    return cached(*args, **kwds)
  File "/Users/mitsuhiko/.pyenv/versions/3.8.12/lib/python3.8/typing.py", line 330, in __hash__
    return hash((self._name,))
RecursionError: maximum recursion depth exceeded while calling a Python object
```

Reply:
I think this is just that it finally failed recursion on that level, but given that `cast()` is just unnecessary runtime overhead, I removed it.